### ISSUE TITLE
No Next/Previous Page navigation after update to 5.16.9

### ISSUE THREAD

**Issue by ppascher** [honestbleeps/Reddit-Enhancement-Suite#5123]
- My browser: Firefox ESR 60.7.0
- My RES version: 5.16.9
- Am I in reddit beta (yes/no):  no

*What's the problem?*
After updating to 5.16.9 I do not have any page navigation elements anymore (previous/next at bottom of page). Possibly because of commit https://github.com/honestbleeps/Reddit-Enhancement-Suite/commit/9a890d37f5ae8843fe589f0dca359e634f3be590
I am using old.reddit.com and have never ending reddit disabled. I tried clearing the res cache and also resetting to factory default which didnt help. Also reset all cookies. Never Ending Reddit does not seem to work either. Downgrading to 5.16.7 fixes this.

*Are there any related posts in https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues or /r/RESissues?*
None that I could find. 

*What other browser extensions are installed?*
uBlock Origin, uMatrix, https everywhere, vim vixen, keepassxc, redirect amp to html

**Comment by ppascher** [comment]
Just tested at home with the same addon setup but on linux instead of windows and Firefox Nightly 69.0a1 (2019-05-29) instead of ESR. Same problem.

**Comment by larsjohnsen** [comment]
Can you confirm that you have neverEndingReddit disabled? I cannot see how the function which hides the navigation buttons would be invoked if that is the case.

**Comment by ppascher** [comment]
I just tried enabling neverEndingReddit, saving, disabling it again and saving. Neither does neverEndingReddit work when enabled nor do the previous/next links reappear after disabling it for me. This was the same as with Firefox ESR at work earlier. Downgrading RES lets me use both.

**Comment by ppascher** [comment]
Solved: After I found out neverendingreddit didnt work (usually have it deactivated) I found this thread: https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues/4781
I noticed page navigation worked when logged out. According to that thread and the link to the wiki I had to allow 3rd party cookies which I did by setting user_pref("network.cookie.cookieBehavior", 4); (was set to 1= Block all 3rd party cookies).
Not sure why it suddenly stopped. Maybe I had an old cookie left.
Sorry for any trouble caused.

**Comment by AnonQuesting** [comment]
I disagree with closing this issue, I've encountered the problem myself also. The fact that not blocking third party cookies "solves" the problem is not fine. Upgrading to 5.16.9 from 5.16.7 still causes a breakage. If this ends up being a "won't fix" that's fine I guess but I thought I'd better voice my opinion nonetheless.

**Comment by honestbleeps** [comment]
@AnonQuesting what browser are you using, and what version number?

**Comment by ppascher** [comment]
My "solution" didn't work anymore the next day even if set to allow all "0". Looking into it further I noticed there were IntersectionObserver related errors.
My user.js from ghacks includes the following:
```
/* 2426: disable Intersection Observer API [FF53+]
 * Almost a year to complete, three versions late to stable (as default false),
 * number #1 cause of crashes in nightly numerous times, and is (primarily) an
 * ad network API for "ad viewability checks" down to a pixel level
 * [1] https://developer.mozilla.org/docs/Web/API/Intersection_Observer_API
 * [2] https://w3c.github.io/IntersectionObserver/
 * [3] https://bugzilla.mozilla.org/1243846 ***/
user_pref("dom.IntersectionObserver.enabled", false);
```
Enabling this fixes the prev/next page missing problem. This entry was added to the user.js in April 2017 already though and hasn't caused this issue before. From what I understand it is needed vor NER so not sure which commit caused issues with NER disabled all of a sudden or if it should be considered a bug.

@AnonQuesting Do you have this preference set to false as well by any chance?

**Comment by AnonQuesting** [comment]
@honestbleeps I'm on Firefox 68.0b6.

@ppascher toggling dom.IntersectionObserver.enabled to true fixes this for me as well.
